yjit/src/asm/mod.rs

```diff
@@ -463,6 +463,12 @@ impl CodeBlock {
         self.dropped_bytes
     }
 
+    /// To patch code that straddle pages correctly, we need to start with
+    /// the dropped bytes flag unset so we can detect when to switch to a new page.
+    pub fn reset_dropped_bytes_for_patching(&mut self) {
+        self.dropped_bytes = false;
```

Comment:
I think we should restore the original flag when it restores the original position with `cb.set_pos(old_pos)`.

Reply:
Ok, I updated it to do that. It's probably fine either way as long as we check the flag as soon as we call `asm.compile`, but restoring it feels safer.